R-package/R/xgb.train.R

```diff
@@ -148,17 +148,43 @@
 #' @param xgb_model a previously built model to continue the training from.
 #'        Could be either an object of class \code{xgb.Booster}, or its raw data, or the name of a
 #'        file with a previously saved model.
+#' @param training_continuation when passing `xgb_model`, whether to update the previous model by
+#'        creating a copy of it which will contain the new boosted rounds (meaning: the original object
+#'        is kept as it was before the call to `xgb.train`, and there will be two booster objects), or
+#'        by updating the previous object in-place (meaning: the object passed under `xgb_model` will
+#'        be updated, and nothing will be returned from this function).
+#'
+#'        Note that, if passing "update" here, the object in `xgb_model`
+#'        \bold{will get updated regardless of whether this function succeeds or not}
+#'        (for example, the parameters will be set on the existing `xgb_model`, overwriting previous ones,
+#'        even if the training fails because of some error).
 #' @param callbacks a list of callback functions to perform various task during boosting.
 #'        See \code{\link{callbacks}}. Some of the callbacks are automatically created depending on the
 #'        parameters' values. User can provide either existing or their own callback methods in order
 #'        to customize the training process.
+#'
+#'        Note that some callbacks might try to set an evaluation log - in order to keep such logs,
+#'        it's necessary to pass `keep_extra_attributes = TRUE`. Be aware that these evaluation logs
+#'        are kept as R attributes, and thus do not get saved when using non-R serializaters like
+#'        \link{xgb.save} (but are kept when using R serializers like \link{saveRDS}).
 #' @param ... other parameters to pass to \code{params}.
 #' @param label vector of response values. Should not be provided when data is
 #'        a local data file name or an \code{xgb.DMatrix}.
 #' @param missing by default is set to NA, which means that NA values should be considered as 'missing'
 #'        by the algorithm. Sometimes, 0 or other extreme value might be used to represent missing values.
 #'        This parameter is only used when input is a dense matrix.
 #' @param weight a vector indicating the weight for each row of the input.
+#' @param keep_extra_attributes Whether to keep extra R attributes in the booster object which
```

Comment:
is this still necessary after your refactor for callbacks? Can we keep callback-specific attributes to callbacks themselves instead of the booster? (a question instead of a request).

Reply:
It's not necessary, but a user that is aiming for minimal footprint (e.g. when serving the model) might want not to add any of these.

Reply:
Do we need to concern about these attributes when serving the model?

Reply:
No, they are not used in `predict` nor importance nor plotting, which is why I wanted to make them optional.

But on second thought, it should be pretty straightforward for a user to remove them, so I've removed this parameter.